lib/core/Observer.ts

```diff
@@ -340,6 +343,10 @@ export default class Observer {
 
     await this.unresolvableTransactionStore.removeUnresolvableTransactionsLaterThan(bestKnownValidRecentTransactionNumber);
 
+    for (const transaction of await this.transactionStore.getTransactionsLaterThan(bestKnownValidRecentTransactionNumber, undefined)) {
```

Comment:
Can we do this by calling one method and pass in the transaction time? Something like `this.confirmationStore.removeConfirmationsLaterThan(bestKnownValidRecentTransactionTime)`?

Reply:
That may be complicated. To do that, I'll need to add transaction time to the data model. And instead of removing the entry, I need to reset the confirmedAt field. Since the reorg will change the transactionTime, I will need to update the transactionTime as well.